packages/wallet/WalletManager.js

```diff
@@ -0,0 +1,421 @@
+    }
+    return this.seed;
+  }
+
+  /**
+   * @return {string} the current seed
+   */
+  getSeed() {
+    if (this.seed == null) {
+      console.log('The seed of the wallet is not set.');
+      return null;
+    }
+    return this.seed;
+  }
+
+  /**
+   * Generate the wallet based on the seed
+   */
+  setWallet() {
+    switch (this.network) {
+      case 'stellar':
+        this.wallet = stellarHDWallet.fromMnemonic(this.seed);
+        break;
+      case 'ethereum':
+        this.wallet = hdkey.fromMasterSeed(this.seed);
+        break;
+      default:
+        this.wallet = null;
+        console.log('The seed is not specified or not valid.');
+    }
+    return this.wallet;
+  }
+
+  /**
+   * @returns {Object} the current wallet
+   */
+  getWallet() {
+    if (this.wallet == null) {
+      console.log('The wallet is not set.');
+      return null;
+    }
+    return this.wallet;
+  }
+
+  /**
+   * Before generating accounts, the seed and wallet MUST be there.
+   * i.e., random account generatation is NOT allowed.
+   * If the wallet is set, generate the first (0) account in the wallet;
+   * If the parameter is a number,
+   * Generate the account in the wallet. Its index is the number.
+   * If the parameter is a string,
+   * Use it as the private/public key to generate the account
+   * @returns {Account} the account created/imported
+   */
+  async getAccount() {
+    if (arguments.length === 0) {
+      if (this.wallet == null) {
+        console.log('The wallet is not initialized.');
+        return null;
+      }
+      // generate the 0th account in the wallet
+      switch (this.network) {
+        case 'stellar': {
+          this.account = new Account(this.network);
+          await this.account.setAccount(this.wallet.getKeypair(0));
+          this.accountArray.push(Object.assign({}, this.account));
+          return this.account;
+        }
+        case 'ethereum': {
+          const privateKey = `0x${this.wallet.deriveChild(0).getWallet()._privKey.toString('hex')}`;
+          this.account = new Account(this.network);
+          await this.account.setAccount(web3.eth.accounts.privateKeyToAccount(privateKey));
+          this.accountArray.push(Object.assign({}, this.account));
+          return this.account;
+        }
+        default:
+          console.log('The network is not set.');
+          return null;
+      }
+    }
+    if (arguments.length === 1 && Number.isInteger(arguments[0]) && arguments[0] >= 0) {
+      // generate the account of the wallet at the specified index
+      switch (this.network) {
+        case 'stellar': {
+          this.account = new Account(this.network);
+          await this.account.setAccount(this.wallet.getKeypair(arguments[0]));
+          this.accountArray.push(Object.assign({}, this.account));
+          return this.account;
+        }
+        case 'ethereum': {
+          const privateKey = `0x${this.wallet.deriveChild(arguments[0]).getWallet()._privKey.toString('hex')}`;
```

Comment:
What if `this.wallet` is null? It is missing the null check at [L163](https://github.com/rate-engineering/rate3-monorepo/pull/33/files#diff-8d78c1de2f3473b830c5024c30c5bf23R163)

Reply:
This null check is moved to the start of the method because all the cases below use methods of `this.wallet`